pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -89,18 +101,34 @@ const CryptoPolicyDialog = ({
     setCurrentCryptoPolicy,
 }) => {
     const [error, setError] = useState();
+    const [inProgress, setInProgress] = useState(undefined);
     const [selected, setSelected] = useState(currentCryptoPolicy);
+    const [rebootRequired, setRebootRequired] = useState(currentCryptoPolicy);
 
     const setPolicy = (reboot) => {
+        const applyPolicy = reboot => {
+            if (reboot) {
+                cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                        .catch(error => setError(error));
+            } else {
+                cockpit.file(applyNeedRebootStamp, { superuser: "require" }).replace("\n")
+                        .then(() => close());
+            }
+        };
         cockpit.spawn(["update-crypto-policies", "--set", selected], { superuser: "require", err: "message" })
                 .then(() => {
                     setCurrentCryptoPolicy(selected);
-                    if (reboot) {
-                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
-                                .catch(error => setError(error));
+                    if (selected === "FIPS") {
+                        setRebootRequired(reboot);
+                        setInProgress(true);
```

Comment:
Let's make this a bit more robust -- move the setInProgress(true) in front of the `update-crypto-policies` call, just in case this takes a non-trivial amount of time (it might change implementation in the future).